src/daft-connect/src/translation/logical_plan/local_relation.rs

```diff
@@ -86,14 +84,24 @@ pub fn local_relation(plan: spark_connect::LocalRelation) -> eyre::Result<Plan>
             })
             .collect();
 
-        let schema = arrow2::datatypes::Schema::from(arrow_fields);
+        let daft_fields: Vec<_> = arrow_fields
+            .iter()
+            .map(daft_schema::field::Field::from)
+            .collect();
+
+        let daft_schema =
+            daft_schema::schema::Schema::new(daft_fields).wrap_err("Could not create schema")?;
+
+        let daft_schema = Arc::new(daft_schema);
+
+        let arrow_schema = arrow2::datatypes::Schema::from(arrow_fields.clone());
```

Comment:
there's a bunch of extra allocations here. We should only need to create the arrow_schema, and the daft_schema. the daft field can be created in the loop instead

```rs
let arrow_schema = arrow2::datatypes::Schema::from(arrow_fields.clone());
let daft_schema = Arc::new(daft_schema::schema::Schema::try_from(&arrow_schema)?);

// ...


for (array, arrow_field) in itertools::zip_eq(chunk, arrow_fields) {
    let daft_field = daft_schema::field::Field::from(arrow_field);
    let field_ref = Arc::new(daft_field);
    let series = Series::from_arrow(field_ref, array)
        .wrap_err("Failed to create Series from Arrow array.")?;

    num_rows.push(series.len());
    columns.push(series);
}

```